skrub/_reporting/_table_report.py

```diff
@@ -134,31 +140,38 @@ def __init__(
         self.dataframe = dataframe
         self.verbose = verbose
         self.max_plot_columns = max_plot_columns
+        self.max_association_columns = max_association_columns
+        self.n_columns = sbd.shape(self.dataframe)[1]
+
+        if self.max_plot_columns is None:
+            self.max_plot_columns = self.n_columns
+        if self.max_association_columns is None:
+            self.max_association_columns = self.n_columns
```

Comment:
I would rather still store them as `None` so it is easy to check if the report was created with max_plot_columns set to None or to the number of columns in the dataframe